core/src/window_service.rs

```diff
@@ -121,25 +123,50 @@ where
     let mut shreds = shred_receiver.recv_timeout(timer)?;
 
     while let Ok(mut more_shreds) = shred_receiver.try_recv() {
-        shreds.append(&mut more_shreds)
+        shreds.append(&mut more_shreds);
     }
 
-    blockstore.insert_shreds_handle_duplicate(
-        shreds,
-        Some(leader_schedule_cache),
-        false,
-        &handle_duplicate,
-        metrics,
-    )?;
+    let (data_shreds, coding_shreds, more_coding_shreds) = blockstore
+        .insert_shreds_handle_duplicate(
```

Comment:
hmmm what if instead of waiting for insert to finish here we did

In `recv_window`, before we send the shreds to insert:
1) iterate over the shreds and build a HashSet of slots
2) For each slot in the hashset call `blockstore.index_cf
            .get(slot)
            .unwrap()
            .unwrap_or_else(|| Index::new(slot))`; 

3) Don't retransmit the ones that already have a slot/index existing in the Index for that slot. 


This should also save the recopying of the shred into packets here.